src/ctia/task/check_es_stores.clj

```diff
@@ -0,0 +1,153 @@
+    [conn :as conn]
+    [document :as es-doc]
+    [index :as es-index]]
+   [clojure.string :as string]
+   [clojure.tools.logging :as log]
+   [ctia
+    [init :refer [init-store-service! log-properties]]
+    [properties :as p :refer [properties]]
+    [store :refer [stores]]]
+   [ctia.entity.entities :refer [entities]]
+   [ctia.entity.sighting.schemas :refer [StoredSighting]]
+   [ctia.stores.es.crud :refer [coerce-to-fn]]
+   [ctia.task.migrations :refer [available-migrations]]
+   [schema-tools.core :as st]
+   [schema.core :as s]))
+
+(def default-batch-size 100)
+(def timeout (* 5 60000))
+
+(def all-types
+  (assoc (apply merge {}
+                (map (fn [[_ {:keys [entity stored-schema]}]]
+                       {entity stored-schema}) entities))
+         :sighting (st/merge StoredSighting
+                             {(s/optional-key :observables_hash) s/Any})))
+
+(defn type->schema [type]
+  (if-let [schema (get all-types type)]
+    schema
+    (do (log/warnf "missing schema definition for: %s" type)
+        s/Any)))
+
+(defn setup
+  "init properties, start CTIA and its store service"
+  []
+  (log/info "starting CTIA Stores...")
+  (p/init!)
+  (log-properties)
+  (init-store-service!))
+
+(defn fetch-batch
+  "fetch a batch of documents from an es index"
+  [{:keys [conn
+           indexname
+           mapping]} batch-size offset sort-keys]
+  (let [params
+        (merge
+         {:offset (or offset 0)
+          :limit batch-size}
+         (when sort-keys
+           {:search_after sort-keys}))]
+    (es-doc/search-docs
+     conn
+     indexname
+     mapping
+     nil
+     {}
+     params)))
+
+(defn check-store
+  "check a single store"
+  [store
+   batch-size]
+  (let [target-store (store->map
+                      store
+                      {:cm (conn/make-connection-manager
+                            {:timeout timeout})})
+        store-schema (type->schema (keyword (:type target-store)))
+        coerce! (coerce-to-fn [store-schema])
+        store-size (-> (fetch-batch target-store 1 0 nil)
+                       :paging
+                       :total-hits)]
+    (log/infof "%s - store size: %s records"
+               (:type target-store)
+               store-size)
+    (loop [offset 0
+           sort-keys nil
+           checked-count 0]
+      (let [{:keys [data paging]
+             :as batch}
+            (fetch-batch target-store
+                         batch-size
+                         offset
+                         sort-keys)
+            next (:next paging)
+            offset (:offset next 0)
+            search_after (:sort paging)
+            checked (coerce! data)
+            checked-count (+ checked-count
+                             (count checked))]
+        (if next
+          (recur offset search_after checked-count)
+          (log/infof "%s - finished checking %s documents"
+                     (:type target-store)
+                     checked-count))))))
```

Comment:
To make the code slightly more modulable I would use a higher order function like: `(do-all-documents store batch-size f)` where `f` will be your function to check each doc. It appears this is the kind of thing that could be reused for further migration process.

Also the `coerce!` is somehow hidden inside the make the `loop` logic. That made a bit difficult for me to discover where the check was really made.